docs/docs/spec.md

```diff
@@ -273,30 +305,31 @@ If another client attempts to initialize a repository in the same location, only
 If the specific snapshot ID is known, a client can open it directly in read only mode.
 
 1. Use the specified snapshot ID to fetch the snapshot file.
-1. Fetch desired attributes and values from arrays.
+1. Inspect the snapshot to find the relevant manifest or manifests.
+1. Fetch the relevant manifests and the desired chunks pointed by them.
 
 #### From Branch
 
 Usually, a client will want to read from the latest branch (e.g. `main`).
 
-1. List the object store prefix `refs/branch.$BRANCH_NAME/` to obtain the latest branch file in the sequence. Due to the encoding of the sequence number, this should be the _first file_ in lexicographical order.
-1. Read the branch file JSON contents to obtain the snapshot ID.
+1. Resolve the object store prefix `refs/branch.$BRANCH_NAME/ref.json` to obtain the latest ref file.
+1. Parse the branch file JSON contents to obtain the snapshot ID.
 1. Use the snapshot ID to fetch the snapshot file.
-1. Fetch desired attributes and values from arrays.
+1. Fetch the relevant manifests and the desired chunks pointed by them.
 
 #### From Tag
 
 1. Read the tag file found at `refs/tag.$TAG_NAME/ref.json` to obtain the snapshot ID.
 1. Use the snapshot ID to fetch the snapshot file.
-1. Fetch desired attributes and values from arrays.
+1. Fetch the relevant manifests and the desired chunks pointed by them.
 
 ### Write New Snapshot
 
 1. Open a repository at a specific branch as described above, keeping track of the sequence number and branch name in the session context.
 1. [optional] Write new chunk files.
 1. [optional] Write new chunk manifests.
 1. Write a new snapshot file.
-1. Attempt to write the next branch file in the sequence
+1. Do conditional update to write the new value of the branch reference file
```

Comment:
Should this section mention writing transaction logs?

Reply:
Did the parts of transaction logs, diff and conflict resolution can come on a second pass